Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp

```diff
@@ -187,6 +187,31 @@ bool Texture::IsCompressed() const
            format_ == COMPRESSED_RGB_PVRTC_2BPPV1_IMG || format_ == COMPRESSED_RGBA_PVRTC_2BPPV1_IMG;
 }
 
+unsigned Texture::GetDataSize(int width, int height) const
+{
+    if (IsCompressed())
+    {
+        if (format_ == COMPRESSED_RGB_PVRTC_4BPPV1_IMG ||
+            format_ == COMPRESSED_RGBA_PVRTC_4BPPV1_IMG ||
+            format_ == COMPRESSED_RGB_PVRTC_2BPPV1_IMG ||
+            format_ == COMPRESSED_RGBA_PVRTC_2BPPV1_IMG)
+        {
+            const unsigned bpp = (format_ == COMPRESSED_RGB_PVRTC_2BPPV1_IMG ||
+                                format_ == COMPRESSED_RGBA_PVRTC_2BPPV1_IMG)?2:4;
+            
+            const unsigned xSize =  (bpp == 2) ? 8 : 4;
+            const unsigned blockSize = 8;
+            // For MBX don't allow the sizes to get too small
+            unsigned blocksWide = Max(2, width / xSize);
+            unsigned blocksHeight = Max(2, height / 4);
+            return blocksWide * blocksHeight * blockSize;
```

Comment:
Code block above is noop, it does not affect result of this function in any way.

Reply:
It does affect the result. May I ask you to elaborate?

Reply:
NVM, i indeed overlooked `return` statement, sorry. 👀